examples/inspection/plot_linear_model_coefficient_interpretation.py

```diff
@@ -0,0 +1,274 @@
+# The dataset is composed by columns with different data types and we need to
+# apply a specific preprocessing for each data types.
+# Our pre-processor will
+#
+# - one-hot encode (i.e., generate a column by category) the categorical
+#   columns;
+# - replace by 0 and 1 the categories of binary columns;
+# - keep numerical values as they are.
+
+from sklearn.compose import make_column_transformer
+from sklearn.preprocessing import OneHotEncoder
+from sklearn.preprocessing import OrdinalEncoder
+
+categorical_columns = ['RACE', 'OCCUPATION', 'SECTOR']
+binary_columns = ['MARR', 'UNION', 'SEX', 'SOUTH']
+numerical_columns = ['EDUCATION', 'EXPERIENCE', 'AGE']
+
+preprocessor = make_column_transformer(
+    (OneHotEncoder(), categorical_columns),
+    (OrdinalEncoder(), binary_columns),
+    remainder='passthrough'
+)
+
+##############################################################################
+# To describe the dataset as a linear model we choose to use a ridge regressor
+# and to model le log of the "WAGE".
+
+from sklearn.pipeline import make_pipeline
+from sklearn.linear_model import RidgeCV
+from sklearn.compose import TransformedTargetRegressor
+
+model = make_pipeline(
+    preprocessor,
+    TransformedTargetRegressor(
+        regressor=RidgeCV(),
+        func=np.log10,
+        inverse_func=sp.special.exp10
+    )
+)
+
+##############################################################################
+# Processing the dataset
+# ......................
+#
+# First of all we split the sample in a train and a test dataset.
+
+from sklearn.model_selection import train_test_split
+
+X_train, X_test, y_train, y_test = train_test_split(
+    X, y, random_state=42
+)
+
+##############################################################################
+# Then we fit the model
+
+model.fit(X_train, y_train)
+
+##############################################################################
+# We can check the performance of the computed model using, for example, the
+# median absolute error of the model.
+
+from sklearn.metrics import median_absolute_error
+
+y_pred = model.predict(X_train)
+mae = median_absolute_error(y_train, y_pred)
+string_score = 'MAE on training set: {0:.2f} $/hour'.format(mae)
+y_pred = model.predict(X_test)
+mae = median_absolute_error(y_test, y_pred)
+
+string_score += '\nMAE on testing set: {0:.2f} $/hour'.format(mae)
+
+fig, ax = plt.subplots(figsize=(6, 6))
+sns.regplot(y_test, y_pred)
+
+plt.text(3, 20, string_score)
+
+plt.ylabel('Model predictions')
+plt.xlabel('Truths')
+plt.xlim([0, 27])
+plt.ylim([0, 27])
+
+##############################################################################
+# The model learnt is far to be a good model making accurate prediction.
+# As interpretation tools characterize model rather than the generative process
+# of the data itself, it needs to be emphasized that interpretations are
+# correct if the model is correct as well.
```

Comment:
Being the devil's advocate here: so why are we trying to interpret the coeff of a bad model?

Reply:
You are right. I have added a comment here: the worst the model the better the methodological  discussion about how do not get trapped in a bad explanation... :) WDYT?

Reply:
I'm not convinced honestly ;)

Let me suggest

> The model learnt is far from being a good model making accurate predictions: the R squared score is very low. In the following section, we will interpret the coefficients of the model. While we do so, please keep in mind that any conclusion we way draw will be about the model that we build, rather than about the true (real-world) generative process of the data. As always, we must be careful when interpreting machine learning models.